step-release-vis/src/app/services/environmentService.ts

```diff
@@ -9,6 +9,8 @@ import {Observable, of} from 'rxjs';
 })
 export class EnvironmentService {
   BEGINNING: number;
+  /* Edges stores the number of points shared by candidates, DTI: edges[x-y] = edges[y-x] */
+  edges: Map<string, number> = new Map();
```

Comment:
this is error prone to candidate names including "-" I would prefer to use a pair<string;string> as a key here, if possible

Reply:
objects in this Map structure are considered equal if their reference is the same, but i can create an class with a hashing function and store the hashed values to avoid this, if it seems fine to you.

Reply:
before this version I was storing a Map<[candidate1, candidate2], number> but if a = [1, 2] and b=[1,2] the Map would not consider a === b.

Reply:
I am fine submitting this as it is, but this needs a fix later, pls create an issue, and add a TODO here for later ;)